lib/relaton/index/type.rb

```diff
@@ -61,7 +61,7 @@ def add_or_update(id, file)
       #
       def search(id = nil)
         index.select do |i|
-          block_given? ? yield(i) : i[:id].include?(id)
+          block_given? ? yield(i) : i[:id] == id
```

Comment:
1. Other gems still use String IDs and rely on `#include?` behavior. We should keep using `#include?` for String ID arguments.
2. `i[:id]` is a Hash, right? So the comparison `i[:id] == id` has Hash on the left and Pubid on the right, correct? How does it work if the Pubid lacks some parts? Sometimes we need to ignore missed parts, sometimes we don't. How can we handle such cases?

Reply:
> 1. Other gems still use String IDs and rely on `#include?` behavior. We should keep using `#include?` for String ID arguments.

I'll update minor version for `relaton-index`, so only relaton gems using pubid will use updated version with these changes.
As I understand you want `#include?` for partial matching, but for pubid identifiers we will do it another way.

> 2. `i[:id]` is a Hash, right? So the comparison `i[:id] == id` has Hash on the left and Pubid on the right, correct? How does it work if the Pubid lacks some parts? Sometimes we need to ignore missed parts, sometimes we don't. How can we handle such cases?

@andrew2net Currently, I'm implementing partial matching in `relaton-ccsds` gem, but it can be implemented in `relaton-index` as well for some generic cases.

Reply:
@mico index is serialized as YAML to save it in a file, where does the `i[:id]` become a `Pubid::Core::Identifier::Base object`?
If the 'i[:id]` is a `Pubid::Core::Identifier::Base object` will compare correctly with all types of Pubid?

Reply:
> @mico index is serialized as YAML to save it in a file, where does the `i[:id]` become a `Pubid::Core::Identifier::Base` object? If the `i[:id]` is a `Pubid::Core::Identifier::Base object` will it compare correctly with all types of Pubid?

@andrew2net I'll update the code to serialize and deserialize pubid correctly